### ISSUE TITLE
Option for there to be only one RaySession GUI instance at a time?

### ISSUE THREAD

**Issue by mxmilkiib** [Houston4444/RaySession#205]
If enabled, running the program a second time would maximise/focus the currently running RS main window.

**Comment by Houston4444** [comment]
Yes, I agree. Running parallel daemons makes almost no sense, and I am pretty sure nobody use it.
So, I think that this option should be the default one.

However, maximize/focus the window will be impossible (or really not easy) under Wayland, unfortunately.

**Comment by mxmilkiib** [comment]
Running parallel daemons makes almost no sense, but running parallel frontends might.

**Comment by Houston4444** [comment]
It's a big topic.

A lot of users use RaySession as a ladish studio, launching in the session clients they always need and launching outside others programs (DAW for example). In this case, we could imagine they use one daemon for the studio and another one to switch sessions (as projects). Note that jack client names could be in (rarely) conflict if they use several instances of the same program in both daemons.

> Edit: I guess you could use multiple frontends to recreate the separate "room" JACK graphs LASH offered.

Yes, it could be a feature, in fact, it's already possible to launch raysession as a NSM client, but it is running in the same JACK instance. I think that the best choice for rooms would be the same thing, but the raysession client starting another JACK instance, I think that carla-patchbay works this way, but I am not sure.  In any case, if you start raysession without arguments, it won't connect the GUI to a daemon under NSM.

> If enabled, running the program a second time would maximize/focus the currently running RS main window.

Another possibility is to stop the running GUI, and connect the new one to the running daemon.

**Comment by crshrprt** [comment]
> Running parallel daemons makes almost no sense, but running parallel frontends might.
> 
> Edit: I guess you could use multiple frontends to recreate the separate "room" JACK graphs LASH offered.

I sometimes use 2 RaySession instances on my laptop. One controlling my laptop JACK as netjack2 client, and one for my workstation, running JACK server with netmanager, playing the sound coming from the laptop to the speakers. 
Being able to control everything from a single machine is a really cool RaySession's feature (and probably not very well known by many). :)

**Comment by Houston4444** [comment]
> I sometimes use 2 RaySession instances on my laptop. One controlling my laptop JACK as netjack2 client, and one for my workstation, running JACK server with netmanager, playing the sound coming from the laptop to the speakers.
Being able to control everything from a single machine is a really cool RaySession's feature (and probably not very well known by many). :)

Yes, you use 2 instances with 2 jack servers, and then parallel daemons make sense. Maybe you could use one session as slave of the the other one. For example, you can add `raysession` to the client list (yes, raysession can be a NSM client), with the environment variable making it start with your jack server.

**Comment by crshrprt** [comment]
From the description seems cool, but I'm not sure I understood what you mean. Maybe could be a new example in the manual? :) Anyway, I just wrote that previous comment to say the would be nice not to "limit" RaySession to one instance by default  (or have an option to disable the "limit"). Thanks.

**Comment by Houston4444** [comment]
> From the description seems cool, but I'm not sure I understood what you mean. Maybe could be a new example in the manual? :)

I did not write that in the manual, it is here for a dev reason (make RaySession a NSM client was a code step before to work on network sessions), and I didn't think it could be really useful. I just say it to you in the case it is useful, but probably it is not. To test this, just 'add executable', type 'raysession', then in client properties set the environment variable if needed.

> I just wrote that previous comment to say the would be nice not to "limit" RaySession to one instance by default (or have an option to disable the "limit"). Thanks.

Yes, I understood ;) !
